app/controllers/idv/in_person/usps_locations_controller.rb

```diff
@@ -0,0 +1,57 @@
+require 'json'
+
+module Idv
+  module InPerson
+    class UspsLocationsController < ApplicationController
+      include IdvSession
+      include UspsInPersonProofing
+
+      # get the list of all pilot Post Office locations
+      def index
+        usps_response = []
+        begin
+          usps_response = Proofer.new.request_pilot_facilities
+        rescue Faraday::ConnectionFailed => _error
+          nil
+        end
+
+        render json: usps_response.to_json
+      end
+
+      # save the Post Office location the user selected to the session
+      def update
+        idv_session.applicant ||= {}
+        idv_session.applicant[:selected_location_details] = permitted_params.as_json
+
+        render json: { success: true }, status: :ok
+      end
+
+      # return the Post Office location the user selected from the session
+      def show
+        selected_location = idv_session.applicant&.[](:selected_location_details) || {}
+
+        # camel case keys
+        returned_location = {}
+        selected_location.keys.each do |key|
+          returned_location[key.camelize(:lower)] = selected_location[key]
+        end
```

Comment:
(Non-blocking) Looks like a good use-case for [`Hash#transform_keys`](https://ruby-doc.org/core-2.7.0/Hash.html#method-i-transform_keys) ?

```suggestion
        returned_location = selected_location.transform_keys { |key| key.camelize(:lower) }
```